actionmailbox/app/controllers/action_mailbox/ingresses/amazon/sns_notification.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require "tempfile"
+
+module ActionMailbox
+  module Ingresses
+    module Amazon
+      class SnsNotification
+        def initialize(params)
+          @params = params
+        end
+
+        def subscription_confirmed?
+          confirmation_response.code&.start_with?("2")
+        end
```

Comment:
Using a predicate method to enact a verb leads to surprising exception handling expectations. As with S3 downloading, is subscription confirmation a responsibility of the SNS notification class? What is? Where should confirmation exceptions be handled?

Reply:
@jeremy I don't fully understand this one, sorry - could you elaborate a bit ?